### DISCUSSION TITLE
Can't return an interface

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#12318] [Ideas]
The [interface documentation](https://docs.dagger.io/manuals/developer/interfaces) shows how to define an interface and use it in arguments. But it doesn't show how to use the interface *in the return value*. I tried, but it doesn't seem to be possible.

It would be very useful to make this possible.

Example:

```golang
type Check interface {
 Pass() bool
 Details() string
}
 
type MyCheck struct {}
 
func (c *MyCheck) Pass() bool {
  return true
}
 
func (c *MyCheck) Details() string {
  return "my check always passes!"
}
 
type MyModule struct{}
 
func (m *MyModule) Check() Check {
 return new(MyCheck)
}
```

**Comment by sipsma** [comment]
This is due to the same problem described in this issue: https://github.com/dagger/dagger/issues/6366

The fix is essentially the same as what it would take to support self-API calls, so it's very much worth doing once there's bandwidth, given how much we'd get out of it.
